### ISSUE TITLE
How to trainmy own dataset？(not COCO)

### ISSUE THREAD

**Issue by kike-0304** [IDEA-opensource/DN-DETR#32]
If I use my own dataset, where do I need to change, the AP I get with my own dataset is 0.0?

**Comment by xueyyao** [comment]
I met the problem too, but I found that I did something wrong about my dataset. I believe you can check your data (especially label) first.

**Comment by kike-0304** [comment]
> I met the problem too, but I found that I did something wrong about my dataset. I believe you can check your data (especially label) first.
I can use my data set normally in other methods, but only this method will have errors, which places have you modified when using your own data set?

**Comment by FengLi-ust** [comment]
You can transform your dataset to COCO format or write a new dataset class for your dataset.

**Comment by youarefree123** [comment]
> You can transform your dataset to COCO format or write a new dataset class for your dataset.

Hi, I have recently tried to train other datasets (cityscapes) using for dn_dab_detr.  The size of the training set is only 3000,  we train the model on 4 Nvidia 3090 GPUs, batch size is 16 (4 images each GPU x 4 GPUs) , accordingly I adjusted the learning rate to 2e-4 (twice of the original), however, when training, I found that the model did not converge, so I changed back to 1e-4, the batch size is still 4, however, after 50 epochs, the result is not as good as  baseline  (Faster-RCNN), do I still need to try to modify other hyperparameters? Which hyperparameters will be modified to enable the model to work in the new dataset?

**Comment by FengLi-ust** [comment]
Hey, as our code and dataloader largely follows [DETR](https://github.com/facebookresearch/detr) and [Deformable DETR](https://github.com/fundamentalvision/Deformable-DETR). They have been widely adopted in many custom dataset. you can refer to their issues about how to train your own dataset. For example, in DETR github page, there are many issues about training on the custom dataset, like [this](https://github.com/facebookresearch/detr/issues/448).

For modifications to the code for your own dataset, there are also some suggestions, like [this](https://github.com/facebookresearch/detr/issues/141) and [this](https://github.com/facebookresearch/detr/issues/9).

Hope these suggestions work for you.

**Comment by youarefree123** [comment]
> Hey, as our code and dataloader largely follows [DETR](https://github.com/facebookresearch/detr) and [Deformable DETR](https://github.com/fundamentalvision/Deformable-DETR). They have been widely adopted in many custom dataset. you can refer to their issues about how to train your own dataset. For example, in DETR github page, there are many issues about training on the custom dataset, like [this](https://github.com/facebookresearch/detr/issues/448).
> 
> For modifications to the code for your own dataset, there are also some suggestions, like [this](https://github.com/facebookresearch/detr/issues/141) and [this](https://github.com/facebookresearch/detr/issues/9).
> 
> Hope these suggestions work for you.

Thanks for the detailed reply which helped me a lot.
